### PR TITLE
create faceless_milestone1.md

### PR DESCRIPTION
**Pull request by HeisenbergLin22**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/963

### CODE REVIEW

File: deliveries/faceless_milestone1_delivery.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Milestone Delivery :mailbox:
+
+> Only the GitHub account, which is responsible for the pull request of the accepted application is allowed to submit milestones. Don't remove any of the mandatory parts presented in bold letters or as headlines!
+
+**The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/General-Grants-Program/blob/master/grants/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/HeisenbergLin22/Grants-Program/blob/master/faceless.md
```
Comment (alxs) [inline deliveries/faceless_milestone1_delivery.md:7]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/faceless.md
```

**Comment by lucasvanmol** [comment]
Hi @HeisenbergLin22, I'm currently looking through your deliverable. I noticed your repo (https://github.com/HeisenbergLin22/Faceless_milestone1) did not have a `LICENSE` file. Could you please add one for Apache 2.0, as per the delivery table?

**Comment by lucasvanmol** [comment]
I've just uploaded my initial evaluation which you can see here: https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faceless_1_lucasvanmol.md

Please check my evaluation notes - I've requested some additions. Under "General Notes", the notes about documentation are pretty important, whereas the notes under github best practices are just suggestions, and I'll still accept the delivery whether or not you decide to implement them.

**Comment by HeisenbergLin22** [comment]
> Hi @HeisenbergLin22, I'm currently looking through your deliverable. I noticed your repo (https://github.com/HeisenbergLin22/Faceless_milestone1) did not have a `LICENSE` file. Could you please add one for Apache 2.0, as per the delivery table?

Hey @lucasvanmol , thanks for your comments. Already added the LICENSE file to the [repo](https://github.com/HeisenbergLin22/Faceless_milestone1) now. Will get back to you on your other comments ASAP.

**Comment by HeisenbergLin22** [comment]
> 



> I've just uploaded my initial evaluation which you can see here: https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faceless_1_lucasvanmol.md
> 
> Please check my evaluation notes - I've requested some additions. Under "General Notes", the notes about documentation are pretty important, whereas the notes under github best practices are just suggestions, and I'll still accept the delivery whether or not you decide to implement them.

Hey @lucasvanmol , thanks for your comments. The following is my response to your notes:

## General Notes

### Documentation

I think the documentation in the README is great; it's clear and I had no issues executing the commands outlined in it. However, I have some issues with the documentation inside the code. 
- Firstly, because most documentation code was `ignore`'d, running `cargo test` failed to catch some mistakes that were present in the documentation. I've gone ahead and took the liberty of fixing this myself in this [pull request](https://github.com/HeisenbergLin22/Faceless_milestone1/pull/1), which you can go ahead and merge if you agree with my changes.

### Response: Already merged your PR. Thanks for your comments.

- The documentation inside `faceless-substrate-node/pallets`:  the comments in the code here, for `faceless` and `proof-of-existance` are just generic example comments from the `template` pallet, you should update them to reflect what the pallet is doing. 

### Response: Please check our [repo](https://github.com/HeisenbergLin22/Faceless_milestone1) for the updated comments. We have also removed the `proof-of-existance` code given the fact that it is irrelevant. Thanks for your comments. 

### Github best practices

Since it seems like the repository was uploaded to github especially for this milestone delivery, this may not be a problem if the files are also hosted privately elsewhere, but if development is going to continue on the github repo then I would suggest the following changes:

### Response: Thanks for your comments, but this repo is indeed purely for the purpose of milestone delivery.

**Comment by lucasvanmol** [comment]
Thanks for those updates @HeisenbergLin22, much appreciated. Before accepting the delivery I've made some more changes in [this pull request](https://github.com/HeisenbergLin22/Faceless_milestone1/pull/2); I'm mainly just running `cargo fmt` and `cargo clippy --fix` to fix up the code formatting a bit. Could you take a look? Thanks!

**Comment by HeisenbergLin22** [comment]
> 

Hey @lucasvanmol , thanks for your comments. Already merge your PR.

**Comment by lucasvanmol** [comment]
Thanks @HeisenbergLin22 , I've accepted the delivery and you can see my final evaluation here: https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faceless_1_lucasvanmol.md

I'll also forward your invoice internally

**Comment by HeisenbergLin22** [comment]
> Thanks @HeisenbergLin22 , I've accepted the delivery and you can see my final evaluation here: https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/faceless_1_lucasvanmol.md
> 
> I'll also forward your invoice internally

Cool, thanks a lot.

**Comment by lucasvanmol** [comment]
Hello @HeisenbergLin22 , did you fill out the [invoice form](https://forms.gle/LSRr7PCjBpEbKGh89)? I can't seem to find it.

**Comment by HeisenbergLin22** [comment]
> Hello @HeisenbergLin22 , did you fill out the [invoice form](https://forms.gle/LSRr7PCjBpEbKGh89)? I can't seem to find it.

Hey @lucasvanmol , sorry I guess I forgot to upload the invoice form last time when I filled the form. Please check the updated invoice form. Thanks. Feel free to leave me a message if there is any problem with the updated form.

**Comment by meldien** [comment]
@HeisenbergLin22 I noted on your invoice is the billing date and also the invoice number missing. 
Could you please update the missing parts?
Many thanks. 
Best

**Comment by HeisenbergLin22** [comment]
> @HeisenbergLin22 I noted on your invoice is the billing date and also the invoice number missing. Could you please update the missing parts? Many thanks. Best

Hey @meldien , I've already updated the invoice form with the billing date and invoice number added. Please check it out. Thanks.

**Comment by meldien** [comment]
@HeisenbergLin22 , thank for the updated invoice. It's now ready in our system to be approved for payment.

**Comment by RouvenP** [comment]
hi @HeisenbergLin22 we transferred the payment today. Thanks

**Pull request merged**